packages/devtools_app/lib/src/standalone_ui/api/vs_code_api.dart

```diff
@@ -55,6 +63,47 @@ abstract interface class VsCodeDevice {
   static const jsonPlatformTypeField = 'platformType';
 }
 
+/// This class defines a debug session exposed by the Dart/Flutter extensions in
+/// VS Code (and must match the implementation there).
+///
+/// All changes to this file should be backwards-compatible and use
+/// [VsCodeCapabilities] to advertise which capabilities are available and
+/// handle any changes in behaviour.
+abstract interface class VsCodeDebugSession {
+  String get id;
+  String get name;
+  String? get vmServiceUri;
+
+  /// The mode the app is running in.
+  ///
+  /// These values are defined by Flutter and at the time of writing can include
+  /// 'debug', 'profile', 'release' and 'jit_release'.
+  ///
+  /// This value may be unavailable (`null`) for older SDKs or for Dart/Test
+  /// sessions.
```

Comment:
do we need to worry about this since the version of DevTools that has the vs code sidebar support will never be an "older" SDK?

Reply:
Probably not, I've updated the comment to remove this (but include that it's not available before a session has started since the mode comes in the `flutter.appStarted` event).